### PR TITLE
rgw: change default ops log backend from rados to file

### PR DESCRIPTION
**Pull request by cbodley**
a proposed change to the default backend when `rgw_enable_ops_log` is enabled, based on discussion in https://github.com/ceph/ceph/pull/45691

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by mattbenjamin — APPROVED** [review]
lgtm

File: src/common/options/rgw.yaml.in [hunk 1]
```diff
@@ -1477,6 +1480,7 @@ options:
   desc: File-system path for ops log.
   long_desc: Path to file that RGW will log ops logs to.
   fmt_desc: The file-system path for writing operations logs.
+  daemon_default: /var/log/ceph/ops-log-$cluster-$name.log
```
Comment (cbodley) [inline src/common/options/rgw.yaml.in:1483]: note that i copied this from the `log_file` option which defaults to `/var/log/ceph/$cluster-$name.log`

Comment (cbodley) [reply src/common/options/rgw.yaml.in:1483]: @cfsnyder i'd like to add a blurb about log rotation for `rgw_ops_log_file_path`. can you summarize how cephadm does this, or point me at docs maybe?

File: src/common/options/rgw.yaml.in [hunk 2]
```diff
@@ -1447,10 +1447,13 @@ options:
   type: bool
   level: advanced
   desc: Use RADOS for ops log
-  long_desc: If set, RGW will store ops log information in RADOS.
+  long_desc: If set, RGW will store ops log information in RADOS. WARNING:
+    there is no automation to clean up these log entries, so by default they
+    will pile up without bound. This MUST NOT be enabled unless the admin has
+    a strategy to manage and trim these log entries.
```
Comment (ktdreyer) [inline src/common/options/rgw.yaml.in:1453]:
```suggestion
    a strategy to manage and trim these log entries by deleting objects from
    the rgw.log pool.
```

Comment (cbodley) [reply src/common/options/rgw.yaml.in]: changed to
> a strategy to manage and trim these log entries with `radosgw-admin log rm`.

Comment (ktdreyer) [reply src/common/options/rgw.yaml.in]: thanks!

**Comment by cbodley** [comment]
hmm
```
[ RUN      ] TestRGWLua.OpsLog
unix socket connect error: 2
../src/test/rgw/test_rgw_lua.cc:682: Failure
Value of: unix_socket_client_ended_ok
  Actual: false
Expected: true
[  FAILED  ] TestRGWLua.OpsLog (8808 ms)
```

File: src/common/options/rgw.yaml.in (hunk 1 above)
Comment (cfsnyder) [reply src/common/options/rgw.yaml.in:1483]: Yeah sure @cbodley. cephadm automatically adds logrotate config on the host for all files matching `/var/log/ceph/{fsid}/*.log`. So if the `rgw_ops_log_file_path` config param matches `/var/log/ceph/*.log` (container path), it'll be handled by the cephadm config.

Here's an example of the config it generated on one of my test clusters:

```
root@ceph01:~# cat /etc/logrotate.d/ceph-74016538-b125-11ec-abb9-3daf21fa986b 
# created by cephadm
/var/log/ceph/74016538-b125-11ec-abb9-3daf21fa986b/*.log {
    rotate 7
    daily
    compress
    sharedscripts
    postrotate
        chown 167:167 /var/log/ceph/74016538-b125-11ec-abb9-3daf21fa986b/*.log; killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror cephfs-mirror || pkill -1 -x 'ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror|cephfs-mirror' || true
    endscript
    missingok
    notifempty
    su root root
}
```

As you can see, it rotates daily and keeps 7 days of compressed logs.

Comment (cbodley) [reply src/common/options/rgw.yaml.in:1486]: that's great, thanks

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2022-04-26_01:02:21-rgw-wip-cbodley-testing-distro-default-smithi/, but depends on the fixes in https://github.com/ceph/ceph/pull/46027 so we'll need to merge that first

**Review by cfsnyder — APPROVED** [review]
Nice!
